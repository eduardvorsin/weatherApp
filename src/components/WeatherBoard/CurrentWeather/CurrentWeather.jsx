import PropTypes from 'prop-types';
import { getWeatherStatusByCode } from '../../../utils/utils';
import classes from './CurrentWeather.module.css';

export default function CurrentWeather({
  className,
  weatherCode,
  temperature,
  testId,
}) {
  const weatherStatus = getWeatherStatusByCode(weatherCode);
  return (
    <div
      className={`${classes['current-weather']} ${className}`}
      data-testid={testId}
    >
      <p className={classes['current-weather__temperature']}>
        {`${temperature}Â°`}
      </p>
      <p className={classes['current-weather__status']}>
        {weatherStatus}
      </p>
    </div>
  );
}

CurrentWeather.defaultProps = {
  className: '',
  weatherCode: 0,
  temperature: 0,
  testId: '',
};

CurrentWeather.propTypes = {
  className: PropTypes.string,
  weatherCode: PropTypes.number,
  temperature: PropTypes.number,
  testId: PropTypes.string,
};
