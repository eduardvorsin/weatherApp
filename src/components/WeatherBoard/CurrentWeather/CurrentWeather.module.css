.current-weather {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  max-width: 200px;
}

.current-weather>*:not(:last-child) {
  margin-bottom: 10px;
}

.current-weather__temperature {
  font-size: 40px;
  color: var(--brand-color);
}

.current-weather__windspeed {
  font-size: 20px;
}

.current-weather__status {
  font-weight: 700;
  font-size: 16px;
  color: var(--secondary-color);
}

.current-weather__status,
.current-weather__temperature,
.current-weather__windspeed {
  margin: 0;
}

@media(min-width:577px) {
  .current-weather__temperature {
    font-size: 60px;
  }
}
