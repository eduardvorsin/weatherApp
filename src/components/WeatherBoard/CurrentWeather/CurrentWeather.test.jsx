import { render, screen } from '@testing-library/react';
import CurrentWeather from './CurrentWeather';

describe('CurrentWeather component tests', () => {
  it('renders correctly', () => {
    render(<CurrentWeather
      className="test-class"
      weatherCode={1}
      temperature={21}
    />);

    expect(screen.getByText(/21Â°/)).toBeInTheDocument();
  });

  it('renders without props', () => {
    render(<CurrentWeather
      testId="current-weather"
    />);

    expect(screen.getByTestId('current-weather')).toBeInTheDocument();
  });

  it('the classname is correctly assigned to the element', () => {
    render(<CurrentWeather
      className="test-class"
      testId="current-weather"
    />);

    expect(screen.getByTestId('current-weather')).toHaveClass('test-class');
  });

  it('snapshot with default data', () => {
    render(<CurrentWeather
      testId="current-weather"
    />);

    expect(screen.getByTestId('current-weather')).toMatchSnapshot();
  });

  it('snapshot with mock text', () => {
    render(<CurrentWeather
      weatherCode={1}
      temperature={13}
      testId="current-weather"
    />);

    expect(screen.getByTestId('current-weather')).toMatchSnapshot();
  });
});
