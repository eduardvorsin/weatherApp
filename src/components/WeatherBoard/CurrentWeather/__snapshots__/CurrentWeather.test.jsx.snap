// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather component tests snapshot with default data 1`] = `
<div
  class="current-weather "
  data-testid="current-weather"
>
  <p
    class="current-weather__temperature"
  >
    0°
  </p>
  <p
    class="current-weather__status"
  >
    ясно
  </p>
</div>
`;

exports[`CurrentWeather component tests snapshot with mock text 1`] = `
<div
  class="current-weather "
  data-testid="current-weather"
>
  <p
    class="current-weather__temperature"
  >
    13°
  </p>
  <p
    class="current-weather__status"
  >
    ясно
  </p>
</div>
`;
